crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -795,19 +779,6 @@ impl AuthorityPerEpochStore {
                ?assigned_versions, ?next_version,
                "locking shared objects");
 
-        // Make an iterator to update the last consensus index.
-
-        // Holding _tx_lock avoids the following race:
-        // - we check effects_exist, returns false
-        // - another task (starting from CheckpointExecutor) writes effects,
-        //    and then deletes locks from assigned_shared_object_versions
-        // - we write to assigned_object versions, re-creating the locks that were just deleted
-        // - now it's possible to run a new tx against old versions of the shared objects.
-        let _tx_lock = self.acquire_tx_lock(&transaction_digest).await;
```

Comment:
We do need to be a bit careful here and make sure we weren't relying on this lock for anything else, even inadvertently.

Reply:
I read through the code following this lock and I don't see any reason we still need it. Btw there is one other place we acquire the same lock, which could be removed.

Reply:
I agree the writes originally inside this lock do not need to be protected. In fact hold tx lock on any write-only logic is not very useful.